stats/view_test.go

```diff
@@ -594,6 +584,298 @@ func Test_View_MeasureFloat64_AggregationCount_WindowSlidingTime(t *testing.T) {
 	}
 }
 
+func Test_View_MeasureFloat64_AggregationSum_WindowCumulative(t *testing.T) {
+	k1, _ := tag.NewKey("k1")
+	k2, _ := tag.NewKey("k2")
+	k3, _ := tag.NewKey("k3")
+	agg1 := SumAggregation{}
+	view, err := NewView("VF1", "desc VF1", []tag.Key{k1, k2}, nil, agg1, Cumulative{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type tagString struct {
+		k tag.Key
+		v string
+	}
+	type record struct {
+		f    float64
+		tags []tagString
+	}
+
+	type testCase struct {
+		label    string
+		records  []record
+		wantRows []*Row
+	}
+
+	tcs := []testCase{
+		{
+			"1",
+			[]record{
+				{1, []tagString{{k1, "v1"}}},
+				{5, []tagString{{k1, "v1"}}},
+			},
+			[]*Row{
+				{
+					[]tag.Tag{{Key: k1, Value: "v1"}},
+					newSumData(6),
+				},
+			},
+		},
+		{
+			"2",
+			[]record{
+				{1, []tagString{{k1, "v1"}}},
+				{5, []tagString{{k2, "v2"}}},
+			},
+			[]*Row{
+				{
+					[]tag.Tag{{Key: k1, Value: "v1"}},
+					newSumData(1),
+				},
+				{
+					[]tag.Tag{{Key: k2, Value: "v2"}},
+					newSumData(5),
+				},
+			},
+		},
+		{
+			"3",
+			[]record{
+				{1, []tagString{{k1, "v1"}}},
+				{5, []tagString{{k1, "v1"}, {k3, "v3"}}},
+				{1, []tagString{{k1, "v1 other"}}},
+				{5, []tagString{{k2, "v2"}}},
+				{5, []tagString{{k1, "v1"}, {k2, "v2"}}},
+			},
+			[]*Row{
+				{
+					[]tag.Tag{{Key: k1, Value: "v1"}},
+					newSumData(6),
+				},
+				{
+					[]tag.Tag{{Key: k1, Value: "v1 other"}},
+					newSumData(1),
+				},
+				{
+					[]tag.Tag{{Key: k2, Value: "v2"}},
+					newSumData(5),
+				},
+				{
+					[]tag.Tag{{Key: k1, Value: "v1"}, {Key: k2, Value: "v2"}},
+					newSumData(5),
+				},
+			},
+		},
+		{
+			"4",
+			[]record{
+				{1, []tagString{{k1, "v1 is a very long value key"}}},
```

Comment:
Do we care about the validation of the long values in the scope of this test? This case seems to be a bit out of the place.